packages/lesswrong/server/resolvers/dialogueChecksResolvers.ts

```diff
@@ -0,0 +1,51 @@
+//import { isDialogueParticipant } from "../../components/posts/PostsPage/PostsPage";
+import DialogueChecks from "../../lib/collections/dialogueChecks/collection";
+import { randomId } from "../../lib/random";
+import { augmentFieldsDict } from "../../lib/utils/schemaUtils";
+import {defineMutation, defineQuery} from "../utils/serverGraphqlUtil";
+
+defineMutation({
+  name: "upsertUserDialogueCheck",
+  resultType: "DialogueCheck",
+  argTypes: "(targetUserId: String!, checked: Boolean!)",
+  fn: async (_, {targetUserId, checked}:{targetUserId:string, checked:boolean}, {currentUser, repos}) => {
+    if (!currentUser) throw new Error("No check user was provided")
+    if (!targetUserId) throw new Error("No target user was provided")
+    const existingCheck = await DialogueChecks.findOne({targetUserId, userId: currentUser._id})
+    const id = existingCheck ? existingCheck._id : randomId()
+    const now = new Date()
+    await repos.dialogueChecks.upsertDialogueCheck(id, currentUser._id, targetUserId, checked, now)
```

Comment:
I'd move the `randomId()` call into `upsertDialogueCheck`

Reply:
(rather than passing it into the function)

Reply:
(same with `now`)

Reply:
moveod random. can't move id because it's used in the return below

Reply:
Ah, in that case you should still move it, and just return the entire record from the upsert query with `RETURNING *`.

Reply:
conversation ongoing